js/pages/configuration/sources/source-manager.js

```diff
@@ -51,275 +58,265 @@ define([
-        && authApi.isPermittedReadSource(self.selectedSourceId())) || !config.userAuthenticationEnabled
-        || !self.selectedSourceId();
-    });
-
-    self.isDeletePermitted = ko.pureComputed(function() {
-      return (config.userAuthenticationEnabled && self.isAuthenticated() &&
-        authApi.isPermittedDeleteSource(self.selectedSource().key())) || !config.userAuthenticationEnabled;
-    });
-
-    self.canEdit = ko.pureComputed(function() {
-      return (config.userAuthenticationEnabled && self.isAuthenticated() &&
-        authApi.isPermittedEditSource(self.selectedSourceId())) || !config.userAuthenticationEnabled;
-    });
-
-    self.canSave = ko.pureComputed(function () {
-      return (self.selectedSource() && self.selectedSource().name() && self.selectedSource().key()
-        && self.selectedSource().connectionString() && self.canEdit());
-    });
-
-    self.canDelete = function () {
-      return (self.selectedSource() && self.selectedSource().key() && self.isDeletePermitted());
-    };
-
-    self.canEditKey = ko.pureComputed(function () {
-      return !self.selectedSourceId();
-		});
-
-    self.options.dialectOptions = [
-      { name: 'PostgreSQL', id: 'postgresql' },
-      { name: 'SQL server', id: 'sqlserver' },
-      { name: 'Oracle', id: 'oracle' },
-      { name: 'Amazon Redshift', id: 'redshift' },
-      { name: 'Google BigQuery', id: 'bigquery' },
-      { name: 'Impala', id: 'impala' },
-      { name: 'Microsoft PDW', id: 'pdw' },
-      { name: 'IBM Netezza', id: 'netezza' },
-    ];
-
-    self.sourceCaption = ko.computed(function(){
-      return (self.model.currentSource() == null || self.model.currentSource().key() == null) ? 'New source' :
-        'Source ' + self.model.currentSource().name();
-    });
-
-      self.newSource = function () {
-          self.selectedSource(new Source());
-          self.dirtyFlag(new ohdsiUtil.dirtyFlag(self.selectedSource()));
-      };
+  class SourceManager extends AutoBind(Component) {
+    constructor(params) {
+      super(params);
+      this.config = config;
+      this.model = params.model;
+      this.loading = ko.observable(false);
+      this.dirtyFlag = this.model.currentSourceDirtyFlag;
+      this.selectedSource = params.model.currentSource;
+      this.selectedSourceId = params.model.selectedSourceId;
+      this.options = {};
+      this.isAuthenticated = authApi.isAuthenticated;
+
+      this.hasAccess = ko.pureComputed(() => {
+        if (!config.userAuthenticationEnabled) {
+          return false;
+        } else {
+          return (
+            config.userAuthenticationEnabled && this.isAuthenticated()
+            && authApi.isPermittedEditConfiguration()
+          )
+          || !config.userAuthenticationEnabled;
+        }
+      });
 
-      function isImpalaDS() {
-          return self.selectedSource() && self.selectedSource().dialect() === 'impala';
-      }
+      this.canReadSource = ko.pureComputed(() => {
```

Comment:
Following construct is very common. Why not to move it to separate function (which would return function)?
`(config.userAuthenticationEnabled && this.isAuthenticated() && SOME PERMISSION CHECK || !config.userAuthenticationEnabled)`